### PR TITLE
fix parsing of attribute groups

### PR DESCRIPTION
**Pull request by jmesnil**

### CODE REVIEW

File: controller/src/test/java/org/jboss/as/controller/persistence/PersistentResourceXMLParserTestCase.java [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+                    )
+                    .build();
+        }
+
+        @Override
+        public void readElement(XMLExtendedStreamReader reader, List<ModelNode> list) throws XMLStreamException {
+            xmlDescription.parse(reader, PathAddress.EMPTY_ADDRESS, list);
+        }
+
+        @Override
```
Comment (ctomc) [inline controller/src/test/java/org/jboss/as/controller/persistence/PersistentResourceXMLParserTestCase.java]: no need for this anymore, you should rather extend PersistentResourceXMLParser, where read/write Element methods are hidden.

Comment (ctomc) [inline controller/src/test/java/org/jboss/as/controller/persistence/PersistentResourceXMLParserTestCase.java]: take a look at IOSubsystemParser_1_1 for example

Comment (jmesnil) [inline controller/src/test/java/org/jboss/as/controller/persistence/PersistentResourceXMLParserTestCase.java]: using `PersistentResourceXMLParser` requires to create a `SubsystemMarshallingContext` that itself requires a `XMLExtendedStreamWriter` which can not be created in a simple way using the staxmapper API.

For the purpose of this tests, implementing directly `XMLElementReader` and `XMLElementReader` should be sufficient.

**Comment by bstansberry** [comment]
Thanks!

**Pull request merged**